File: test/ble_connect_source_contract_test.dart
import 'package:flutter_ezw_ble/core/models/ble_connect_model.dart';
import 'package:flutter_ezw_ble/core/models/ble_connect_source.dart';
import 'package:flutter_ezw_ble/core/models/ble_connect_state.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('device-wake reconnect sources survive the Flutter wire model', () {
    for (final source in const <BleConnectSource>[
      BleConnectSource.androidCdm,
      BleConnectSource.androidBlePendingIntent,
    ]) {
      final encoded = BleConnectModel(
        'fixture-id',
        'fixture-name',
        BleConnectState.connected,
        source: source,
      ).toJson();
      expect(encoded['source'], source.name);
      expect(BleConnectModel.fromJson(encoded).source, source);
      expect(source.isAutomaticReconnect, isTrue);
    }
  });
}

File: android/src/test/kotlin/com/fzfstudio/ezw_ble/ble/BleConnectSourceContractTest.kt
package com.fzfstudio.ezw_ble.ble

import com.fzfstudio.ezw_ble.ble.models.BleConnectSource
import kotlin.test.Test
import kotlin.test.assertEquals

class BleConnectSourceContractTest {
    @Test
    fun flutterDeviceWakeSourcesRemainDistinctInNativeAdmissions() {
        assertEquals(
            BleConnectSource.ANDROID_CDM,
            BleConnectSource.fromFlutterValue("androidCdm"),
        )
        assertEquals(
            BleConnectSource.ANDROID_BLE_PENDING_INTENT,
            BleConnectSource.fromFlutterValue("androidBlePendingIntent"),
        )
        assertEquals(true, BleConnectSource.ANDROID_CDM.isAutomaticReconnect)
        assertEquals(true, BleConnectSource.ANDROID_BLE_PENDING_INTENT.isAutomaticReconnect)
    }
}
